test/services/user-group/index.test.js

```diff
@@ -144,6 +144,38 @@ describe('classes service', () => {
 			expect(data[2].displayName).to.equal(classes[1].displayName);
 		});
 
+		it('should display the classes in correct order when years are included', async () => {
+			const { _id } = await testObjects.createTestSchool();
+			const school = await app.service('schools').get(_id);
+			const adminUser = await testObjects.createTestUser({ roles: ['administrator'] });
+
+			const classes = [
+				await testObjects.createTestClass({
+					name: 'A',
```

Comment:
please check longer names and numbers, how are the following elements sorted:
001-abc vs. 1-abc, and ABC vs. abc, ...?

Reply:
I dont really see that as part of the ticket, since the changes are only about adding years to the sorting, and making sure they supercede any ordering by the name.

Also I find it discussable wether that should be in the tests, since such sorting behavior is handled by MongoDB.

To answer the question, numbers come before letters, uppercase before lowercase... so your four examples would appear exactly in that order.